src/components/navigation/Pagination.tsx

```diff
@@ -1,30 +1,49 @@
 import classnames from 'classnames';
-import type { JSX } from 'preact';
+import type { ComponentChildren } from 'preact';
 
-import type { PresentationalProps } from '../../types';
 import { pageNumberOptions } from '../../util/pagination';
 import { ArrowLeftIcon, ArrowRightIcon } from '../icons';
 import Button from '../input/Button';
-import type { ButtonProps } from '../input/Button';
 
-type NavigationButtonProps = PresentationalProps &
-  ButtonProps &
-  Omit<JSX.HTMLAttributes<HTMLButtonElement>, 'icon' | 'size'>;
+type NavigationButtonProps = {
+  children: ComponentChildren;
+  invisible?: boolean;
+  onClick: (e: MouseEvent) => void;
+  pressed?: boolean;
+  title?: string;
+};
 
-function NavigationButton({ ...buttonProps }: NavigationButtonProps) {
+function NavigationButton({
+  children,
+  invisible = false,
+  pressed = false,
+  onClick,
+  title,
+}: NavigationButtonProps) {
   return (
     <Button
       classes={classnames(
         'px-3.5 py-2.5 gap-x-1',
         'font-semibold rounded',
         // These colors are the same as the "dark" variant of IconButton
-        'text-grey-7 bg-grey-2 enabled:hover:text-grey-9 enabled:hover:bg-grey-3',
+        'text-grey-7 enabled:hover:text-grey-9 enabled:hover:bg-grey-3',
         'disabled:text-grey-5 aria-pressed:bg-grey-3 aria-expanded:bg-grey-3',
+
+        // Disabled navigation buttons are rendered as invisible and disabled
+        // rather than removed so that the overall width of the component and
+        // positions of child controls doesn't change too much when navigating
+        // between pages.
+        invisible && 'invisible',
```

Comment:
I reckon this should be equivalent to `{ invisible }`

```suggestion
        { invisible },
```